src/apps/chifra/internal/scrape/handle_scrape_consolidate.go

```diff
@@ -40,10 +40,12 @@ func (opts *ScrapeOptions) HandleScrapeConsolidate(progressThen *rpcClient.MetaD
 		// we need to move the file to the end of the scraped range so we show progress
 		stageFn, _ := file.LatestFileInFolder(stageFolder) // it may not exist...
 		stageRange := base.RangeFromFilename(stageFn)
-		newRange := base.FileRange{First: stageRange.First, Last: blazeOpts.StartBlock + opts.BlockCnt - 1}
-		newFilename := filepath.Join(stageFolder, newRange.String()+".txt")
-		os.Rename(stageFn, newFilename)
-		os.Remove(stageFn) // seems redundant, but may not be on some operating systems
+		if stageRange.Last != (blazeOpts.StartBlock + opts.BlockCnt - 1) {
```

Comment:
I think it is true to say that this could be strictly less than instead of not equal to. In other words, the old staging file had a range that ends earlier than `StartBlock + Count - 1`. If that's the case, I think I'd prefer a less than here. That would make it more explicit.